actionview/lib/action_view/template_eager_loader.rb

```diff
@@ -0,0 +1,126 @@
+module ActionView
+  # = Action View Template Eager Loader
+  class TemplateEagerLoader
+    # ActionView::TemplateEagerLoader finds and caches all
+    # view templates and then compiles them.
+    #
+    # If this is done on server startup,
+    # the first request will be faster.
+
+    def initialize(resolver)
+      @resolver = resolver
+    end
+
+    # Compiles and caches all templates 
+    def eager_load
+      each_template(&:compile!)
+    end
+
+    private
+
+    # Determines all possible template parameters and then
+    # delegates finding and caching to the Resolver.
+    # After finding and caching the templates, it will
+    # yield them to the block.
+    def each_template(&block)
+      prefixes.each do |prefix|
+        path_names(prefix).each do |name|
+          locales.each do |locale|
+            details, key, action, partial, locals = set_template_parameters(locale, name)
+            locals.each do |locals_array|
+              @resolver.find_all(action, prefix, partial, details, key, locals_array).each(&block)
+            end
+          end
+        end
+      end
+    end
+
+    def set_template_parameters(locale, name)
+      name_parts, partial = name_parts_and_partial(name)
+      details, key = details_and_key(locale, name, name_parts)
+      action = partial ? name_parts.first[1..-1] : name_parts.first
+      locals = partial ? locals(action) : [[]]
+      [details, key, action, partial, locals]
+    end
+
+    def name_parts_and_partial(name)
+      name_parts = name.split('.')
+      [name_parts, name_parts.first[0] == '_']
+    end
+
+    def prefixes
+      prefixes = view_paths.map { |name| name.split('app/views/').last }
+      prefixes.map { |prefix| prefix.split(File.basename prefix).first[0..-2] }.uniq
+    end
+
+    def details_and_key(locale, name, name_parts)
+      details = {
+        locale: locale,
+        formats: formats(name),
+        variants: variants(name_parts),
+        handlers: ActionView::Template::Handlers.extensions
+      }
+      [details, ActionView::LookupContext::DetailsKey.get(details)]
+    end
+
+    def locals(action)
+      locals = []
+      locals << action
+      locals << action + '_counter'
+      locals << action + '_iteration'
+      [[], locals]
+    end
+
+    # All possible locale combinations
+    def locales
+      locales = [[:en]]
+      available = I18n.available_locales - [:en]
+      available.each do |locale|
+        locales << [locale, :en]
+        available.each do |default|
+          next if locale == default
+          locales << [locale, :en, default]
+        end
+      end
+      locales
+    end
+
+    def view_paths
+      @view_paths ||= Dir.glob('app/views/**/*.*')
```

Comment:
Very good. Exactly this. The custom paths for views are injected in the Resolver class. We are now using the resolver paths so it should owrk.